### PR TITLE
handle missing metrics values before GP fit

### LINKED ISSUE #436: filter GP data for missing values
For #432 

Before running the GP in `fit_gp()`, remove any missing values and throw a warning before running the GP model.

### PR DESCRIPTION
**Pull request by topepo**
Closes #436

Affects #432

* When the Bayesian optimization data contain missing values, these are removed before fitting the GP model. 
* If all metrics are missing (or the same), no GP is fit and the current results are returned.

### CODE REVIEW

File: R/tune_bayes.R [hunk 1]
```diff
@@ -244,6 +244,8 @@ tune_bayes_workflow <-
       message_wrap(paste("Optimizing", metrics_name, "using", objective$label))
     }
 
+    gp_mod <- NULL
```
Comment (DavisVaughan) [inline R/tune_bayes.R:247]: Probably a little clearer if this is `prev_gp_mod <- NULL`

File: R/tune_bayes.R [hunk 2]
```diff
@@ -252,6 +254,7 @@ tune_bayes_workflow <-
 
       check_time(start_time, control$time_limit)
 
+      prev_gp_mod <- gp_mod
```
Comment (DavisVaughan) [inline R/tune_bayes.R:257]: Then you can move this to the very end of the for loop

File: R/tune_bayes.R [hunk 3]
```diff
@@ -414,14 +419,16 @@ encode_set <- function(x, pset, as_matrix = FALSE, ...) {
...
   dat <-
     dat %>%
     dplyr::filter(.metric == metric) %>%
+    check_gp_data() %>%
     dplyr::select(dplyr::one_of(pset$id), mean)
 
   x <- encode_set(dat %>% dplyr::select(-mean), pset, as_matrix = TRUE)
 
-  if (nrow(x) <= ncol(x) + 1) {
+  if (nrow(x) <= ncol(x) + 1 & nrow(x) > 0) {
```
Comment (DavisVaughan) [inline R/tune_bayes.R:431]:
```suggestion
  if (nrow(x) <= ncol(x) + 1 && nrow(x) > 0) {
```

File: tests/testthat/test-bayes.R [hunk 4]
```diff
@@ -454,3 +454,53 @@ test_that('too few starting values', {
...
+          resamples = folds,
+          initial = 3,
+          metrics = metric_set(rsq),
+          param_info = parameters(cost_complexity(c(-2, 0)))
+        )
+
+    },
+    regexp = NA
+    ),
+    regexp = "removed before fitting the Gaussian"
```
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/test-bayes.R:484]: Agreed.

**Pull request merged**